mk/README.md

```diff
@@ -0,0 +1,152 @@
+├── private.mk.example      # Variable private example file (depends on the project) which
+                            # overrides environment variable
+├── projectdir.mk           # Store at PROJECT_DIR the base directory for the repository
+├── swag.mk
+└── variables.mk            # Default values to the project variables that has not been
+                            # overrided by the environment variables nor by the
+                            # private.mk file
+```
+
+## Quickly usage
+
+- Print out help: `make help`
+- Get dependencies: `make get-deps`
+- Build binaries: `make build`
+- Clean binaries: `make clean`
+- Launch clean tests: `make db-clean clean build db-up test`
+- Prepare database without the migrations: `make db-clean db-up RUN_MIGRATION=`
+- Open the postgres cli in the current executing container: `make db-cli-connect`
+- Populate database with random records: `make db-migrate-seed`
+- Build the docker container image: `make docker-build`
+- Push the docker image to the registry:
+
+  ```sh
+  make docker-build docker-push
+  ```
+
+  > By default it uses the local username as the `QUAY_USER`. Can be overrided.
+  > It uses the image name that is published by default.
+  > It tag the image with the current git short hash.
+  > We can publish in our personal quay.io account, deploy in ephemeral
+  > environment the fresh image container, and see if the image behave
+  > as we expect before to push our changes to the repository.
+
+  ```sh
+  make docker-build docker-push DOCKER_IMAGE="quay.io/my-user/my-image:my-tag"
+  ```
+
+  ```sh
+  make docker-build docker-push QUAY_USER="my-user"
+  ```
+
+  ```sh
+  make docker-build docker-push DOCKER_IMAGE_BASE="quay.io/my-user/my-image"
+  ```
+
+  > See `make printvars` to check related variables.
+
+## More detailed explanation
+
+It uses single responsability principal and split basic or logical groupable
+rules in a single files.
+
+We could group into the following way:
+
+@Minimal and layout composition
+
+- `Makefile`: Entrypoint and the only responsability is to include
+  the list of includes and set the dafault rule.
+- `mk/includes.mk`: This define the composition for your repository
+  and it reference other `mk/*` files, and the `mk/meta-*` files
+  which allow the `help` rule to print out the group title in the
+  right position.
+
+@Variables
+
+- `mk/projectdir.mk`: It uses to be the first include and just store
+  at the `PROJECT_DIR` variable the path to the repository in the
+  file system.
+- `mk/private.mk`: Not used into this project, but it could define
+  by using conditional assignment values that we want to set for
+  our make rules; useful for the specific rules related with our
+  tech-stack, or when building container images, to customize
+  our own registry (quay.io/my-username); provide a mechanism
+  of customization to avoid repeat steps manually as `export MY_VAR=my-value`.
```

Comment:
I think we should either remove the text about private.mk or add support for it in in our project (likely adding it to includes.mk, adding a blank one, and adding it to git ignore?)

Reply:
Thanks! Updated.

Reply:
thanks! I removed the references to `private.mk`; I didn't pay attention to the other parts.